presto-thrift/src/main/java/com/facebook/presto/thrift/ThriftSplitManager.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.ConnectorSplitSource;
+import com.facebook.presto.spi.ConnectorTableLayoutHandle;
+import com.facebook.presto.spi.connector.ConnectorSplitManager;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.thrift.clientproviders.PrestoClientProvider;
+import com.facebook.presto.thrift.interfaces.client.ThriftConnectorSession;
+import com.facebook.presto.thrift.interfaces.client.ThriftPrestoClient;
+import com.facebook.presto.thrift.interfaces.client.ThriftSplitBatch;
+import com.facebook.presto.thrift.interfaces.client.ThriftTableLayout;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.concurrent.MoreFutures.toCompletableFuture;
+import static io.airlift.concurrent.Threads.threadsNamed;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.stream.Collectors.toList;
+
+public class ThriftSplitManager
+        implements ConnectorSplitManager
+{
+    private final PrestoClientProvider clientProvider;
+    private final ThriftClientSessionProperties clientSessionProperties;
+    private final ExecutorService executor;
+
+    @Inject
+    public ThriftSplitManager(PrestoClientProvider clientProvider, ThriftClientSessionProperties clientSessionProperties)
+    {
+        this.clientProvider = requireNonNull(clientProvider, "clientProvider is null");
+        this.clientSessionProperties = requireNonNull(clientSessionProperties, "clientSessionProperties is null");
+        this.executor = newCachedThreadPool(threadsNamed("splits-transform-%s"));
```

Comment:
I looked at the code that uses this pool again and decided to removeI removed it. The pool is used to run a transformation of thrift splits and host addresses which should be trivial enough to run in the same thread.